stl/inc/random

```diff
@@ -500,16 +492,13 @@ _NODISCARD _Uint _Next_linear_congruential_value(_Uint _Prev) noexcept {
         const auto _Mul =
             static_cast<unsigned int>(_Prev) * static_cast<unsigned int>(_Ax) + static_cast<unsigned int>(_Cx);
         return static_cast<_Uint>(_Mul % _Mx);
-    } else if constexpr (_Cx <= ULLONG_MAX && static_cast<_Uint>(_Mx - 1) <= (ULLONG_MAX - _Cx) / _Ax) {
+    } else if constexpr (static_cast<_Uint>(_Mx - 1) <= (ULLONG_MAX - _Cx) / _Ax) {
         // unsigned long long is sufficient to store intermediate calculation
         const auto _Mul = static_cast<unsigned long long>(_Prev) * _Ax + _Cx;
         return static_cast<_Uint>(_Mul % _Mx);
     } else { // no intermediate integral type fits; fall back to multiprecision
-        _MP_arr _Wx;
-        _MP_Mul(_Wx, _Prev, _Ax);
-        _MP_Add(_Wx, _Cx);
-        _MP_Rem(_Wx, _Mx);
-        return static_cast<_Uint>(_MP_Get(_Wx));
+        const auto _Mul = _Unsigned128{_Prev} * _Ax + _Cx;
+        return static_cast<_Uint>(_Mul % _Mx);

```

Comment:
I thing this deserves `performance` label.
You've replaced homegrown big math with `_umul128` / `_udiv128` instrinsics.

Reply:
Oh you're right, the modern implementation is superior!